### PR TITLE
use the upstream authorization filters

### PR DESCRIPTION
**Pull request by deads2k**
This brings in a config option from upstream and uses it to remove custom handler code we had.

@openshift/sig-security

### CODE REVIEW

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -287,7 +287,7 @@ func (c *MasterConfig) buildHandlerChain() (func(apiHandler http.Handler, kc *ap
 
 			// these are all equivalent to the kube handler chain
 			///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-			handler = serverhandlers.AuthorizationFilter(handler, c.Authorizer, c.AuthorizationAttributeBuilder, genericConfig.RequestContextMapper)
+			handler = apifilters.WithAuthorization(apiHandler, c.RequestContextMapper, c.Authorizer)
```
Comment (enj) [inline pkg/cmd/server/origin/master.go]: I think you want `handler`, not `apiHandler`.

File: test/cmd/authentication.sh [hunk 2]
```diff
@@ -47,29 +47,29 @@ os::cmd::expect_success "oc policy can-i --list"
 whoamitoken="$(oc process -f "${OS_ROOT}/test/testdata/authentication/scoped-token-template.yaml" TOKEN_PREFIX=whoami SCOPE=user:info USER_NAME="${username}" USER_UID="${useruid}" | oc create -f - -o name | awk -F/ '{print $2}')"
 os::cmd::expect_success_and_text "oc get user/~ --token='${whoamitoken}'" "${username}"
 os::cmd::expect_success_and_text "oc whoami --token='${whoamitoken}'" "${username}"
-os::cmd::expect_failure_and_text "oc get pods --token='${whoamitoken}' -n '${project}'" "prevent this action; User \"scoped-user\" cannot list pods in project \"${project}\""
+os::cmd::expect_failure_and_text "oc get pods --token='${whoamitoken}' -n '${project}'" "prevent this action; User \"scoped-user\" cannot list pods in the namespace \"${project}\""
```
Comment (enj) [inline test/cmd/authentication.sh]: I feel like this going to break people.  And I am a bit confused why our forbidden message maker is not keeping this from changing.

File: test/cmd/status.sh [hunk 3]
```diff
@@ -44,15 +44,15 @@ os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certi
 os::cmd::expect_success_and_text 'oc status' "You don't have any projects. You can try to create a new project, by running"
 os::cmd::expect_success_and_text 'oc status --all-namespaces' "Showing all projects on server"
 # make sure `oc status` does not re-use the "no projects" message from `oc login` if -n is specified
-os::cmd::expect_failure_and_text 'oc status -n forbidden' 'Error from server \(Forbidden\): User "test-user" cannot get project "forbidden"'
+os::cmd::expect_failure_and_text 'oc status -n forbidden' 'Error from server \(Forbidden\): User "test-user" cannot get projects in the namespace "forbidden"'
```
Comment (enj) [inline test/cmd/status.sh]: This is confusing, to say the least.

File: pkg/cmd/server/origin/master.go (hunk 1 above)
Comment (deads2k) [reply pkg/cmd/server/origin/master.go]: Yikes.  Hopefully that's what failed test-cmd.

File: test/cmd/authentication.sh (hunk 2 above)
Comment (deads2k) [reply test/cmd/authentication.sh]: The filter response writer upstream (what this is using) is shaped this way.  For all but get projects, this is a fair equivalent to have to deal with.

File: test/cmd/status.sh (hunk 3 above)
Comment (deads2k) [reply test/cmd/status.sh]: This particular one is ugly.  I'd rather take the ugly and try to make it prettier upstream.  Having separate handling chains is causing us to keep and track bugs that are already fixed upstream.

**Comment by deads2k** [comment]
updated

**Comment by deads2k** [comment]
got an ok on irc from @enj

**Pull request merged**